lib/Logger.php

```diff
@@ -0,0 +1,35 @@
+<?php
+/**
+ * @package ActiveRecord
+ */
+namespace ActiveRecord;
+
+/**
+ * Defines a logger for logging queries.
+ *
+ * @package ActiveRecord
+ */
+interface Logger
+{
+
+	/**
+	 * Logs a message.
+	 *
+	 * Implementations should expect that `done()` will be called after a call
+	 * to this method, for logged SQL queries.
+	 *
+	 * @param string $message The message to log.
+	 * @param array $params Any parameters associated with the message.
+	 * @return void
+	 */
+	public function log($message, array $params = null);
```

Comment:
Hrm, yeah had forgotten about that. Damn... well I see you did the best you could with the "one word" constraint then.

Given the reach of this isn't too far, no need to press further on it.